src/language-js/comments.js

```diff
@@ -578,7 +578,7 @@ function handleCommentAfterArrowParams(text, enclosingNode, comment, options) {
     comment,
     options.locEnd
   );
-  if (text.slice(index, index + 2) === "=>") {
+  if (index !== false && text.slice(index, index + 2) === "=>") {
```

Comment:
If we  change  non-exists index  to  `-1` (like `String #indexOf`) will that  a  breaking chang ?

Reply:
I don't know. I would actually favor using `-Infinity` to signify non-exists index since it should be safer against a mistake like code adding 1 to a non-exists index. And I would also favor making this kind of change after we get the JSDoc typecheck working on the affected modules.

Reply:
But  in  js  world, `{String,Array}#indexOf` `Array#findIndex` all use  `-1`, I'm okay  with  `false` if it's breaking change .

Reply:
It's public api `src/common/utils-shared.js`, so we can't  change it, unless we  use `-1` internal and  expose  public one  differently, then  make breaking changes  in  `v3`.

Reply:
> use `-1` internal and expose public one differently

Looks like extra complexity not needed  to me. Interesting idea, though.

We could just use `@ts-ignore` comment for now and raise a new PR to fix it with the breaking API changes.

Reply:
Personal I prefer that way, but I think we can keep it your way, and we can change it to `!= -1`  when refactoring.